hnn_core/network_builder.py

```diff
@@ -503,11 +504,19 @@ def _connect_celltypes(self):
                         self.ncs[connection_name].append(nc)
 
     def _record_extracellular(self):
+        from .extracellular import _calc_potentials_callback
```

Comment:
import globally?

Reply:
I you want, sure, but I though it might be OK to hide it here, since it's only used conditionally?

Reply:
From a stylistic perspective, it's preferable to keep the imports all together and in order of more general to specific modules. The only times you want to do nested imports are: 1) to avoid a hard dependency and 2) avoid circular dependency. The circular dependency is a symptom though and it may indicate that we need to introduce a new module/file. E.g., if A depends on B and B on A, move the conflicting function to C (e.g., utils.py) and make both A and B import C. So there is a clear hierarchical structure.

Reply:
Cool, thanks for the elaboration. I've found a way to avoid the need entirely here, though.